{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "87cb7431",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from matplotlib import pyplot as plt\n",
    "from scipy.cluster.hierarchy import dendrogram, linkage\n",
    "from scipy.spatial.distance import pdist\n",
    "import scipy.cluster.hierarchy as sch\n",
    "from scipy.cluster.hierarchy import average,complete,single,weighted,centroid\n",
    "import plotly.figure_factory as ff\n",
    "import scipy.spatial as scs\n",
    "from scipy.spatial.distance import pdist,squareform\n",
    "from numpy import savetxt\n",
    "from matplotlib.ticker import AutoMinorLocator\n",
    "from matplotlib import gridspec\n",
    "import plotly.figure_factory as ff\n",
    "import scipy.cluster.hierarchy as sch\n",
    "import matplotlib\n",
    "import seaborn as sns; sns.set()\n",
    "from scipy.cluster.hierarchy import dendrogram, fcluster, leaves_list,linkage\n",
    "from scipy.spatial import distance\n",
    "from scipy.cluster import hierarchy# You can use SciPy one too\n",
    "import os\n",
    "import sys\n",
    "import warnings\n",
    "import glob\n",
    "from Bio import SeqIO\n",
    "import itertools\n",
    "import numpy as np\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "fafbd4c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "startTime = time.time()\n",
    "# Collect the path for .rgi files from user and load\n",
    "path = '/home/amjad/NICHEnbhd/allrgisrequired/'\n",
    "readfiles=glob.glob(os.path.join(path,\"*.txt\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "b065ebb1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Collect the path for .gbk files from user and load\n",
    "path = '/home/amjad/NICHEnbhd/allgbksrequired/'\n",
    "gbkfiles=glob.glob(os.path.join(path,\"*.gbk\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "2ece2575",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to extract the required data from gbk files(entire genomes:) using biopython library #  \n",
    "def extract(infile):\n",
    "  gene_start=[]\n",
    "  gene_end=[]\n",
    "  gene_strand=[]\n",
    "  gene_name=[]\n",
    "  loc_tag=[]\n",
    "  function=[]\n",
    "  protein_seq=[]\n",
    "  contig_name=[]\n",
    "  unique=[]\n",
    "\n",
    "  for index, record in enumerate(SeqIO.parse(infile, \"genbank\")):\n",
    "    #print(\"index %i, ID = %s, length %i, with %i features\"% (index, record.id, len(record.seq), len(record.features)))\n",
    "    for i in record.features:\n",
    "        if i.type == \"CDS\" and \"gene\" in i.qualifiers:\n",
    "          locations=i.location\n",
    "          gene_start.append(locations.start)\n",
    "          gene_end.append(locations.end)\n",
    "          gene_strand.append(locations.strand)\n",
    "          loc_tag.append(i.qualifiers['locus_tag'])\n",
    "          function.append(i.qualifiers['product'])\n",
    "          protein_seq.append(str(i.qualifiers['translation']))\n",
    "          gene_name.append(i.qualifiers['gene'])\n",
    "          contig_name.append(record.id)\n",
    "        elif i.type ==\"CDS\":\n",
    "          locations=i.location\n",
    "          gene_start.append(locations.start)\n",
    "          gene_end.append(locations.end)\n",
    "          gene_strand.append(locations.strand)\n",
    "          loc_tag.append(i.qualifiers['locus_tag'])\n",
    "          function.append(i.qualifiers['product'])\n",
    "          protein_seq.append(str(i.qualifiers['translation']))\n",
    "          gene_name.append(\"UID\")\n",
    "          contig_name.append(record.id)\n",
    "          \n",
    "  salmonella_gene_frame=pd.DataFrame()\n",
    "  salmonella_gene_frame['GeneStart']=gene_start\n",
    "  salmonella_gene_frame['GeneEnd']=gene_end\n",
    "  salmonella_gene_frame['GeneStrand']=gene_strand\n",
    "  salmonella_gene_frame['Locus_Tag']=loc_tag\n",
    "  salmonella_gene_frame['GeneName']=gene_name\n",
    "  salmonella_gene_frame['Product']=function\n",
    "  salmonella_gene_frame['ProteinSequence']=protein_seq\n",
    "  salmonella_gene_frame['contig_name']=contig_name\n",
    "  \n",
    "  #print(contig_name)\n",
    "  for i in contig_name:\n",
    "    if i not in unique:\n",
    "      unique.append(i)\n",
    "\n",
    "\n",
    "  return salmonella_gene_frame,unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "ecd5fc31",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to create groups based on contigs #\n",
    "def make_groups(frame,column):\n",
    "  group=frame.groupby(frame[column])\n",
    "  datasets = {}  \n",
    "  for groups, data in group:\n",
    "    datasets[groups] = data\n",
    "  return datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "0b78edcc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Creating separate dictionary for all the unique AMR genes of the genome as \"key\" using the field \"Best_Hit_ARO\" #\n",
    "def createeachdict_drug(drugindex):\n",
    "\ttemp_dict={}\n",
    "\tfor j,k in datadict.items():       \n",
    "\t    temp=k[k['Best_Hit_ARO']==drugindex]\n",
    "\t    if len(temp)>0:\n",
    "\t        temp_dict[j]=temp\n",
    "\treturn temp_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "55b30fad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A function to assign color to all and any AMR genes in the neighborhood based on CARD color codes for strict, loose and perfect hits #\n",
    "color_Dict={\"Loose\":\"#DC7633 \",\"Perfect\":\"#28B463\",\"Strict\":\"#F4D03F\"}\n",
    "def checkforRGIinneighborhood(e,genome,rgi_gene):\n",
    "    for q in range(len(e[\"GeneEnd\"])):\n",
    "        for p in range(len(datadict[genome][\"Stop\"])):\n",
    "            datadict[genome].reset_index(drop=True, inplace=True)\n",
    "            #print(e[\"GeneEnd\"][q],p)\n",
    "            if e[\"GeneEnd\"][q] == datadict[genome][\"Stop\"][p] and e[\"GeneName\"][q]==rgi_gene:\n",
    "                e[\"Genecolor\"][q]=color_Dict[datadict[genome][\"Cut_Off\"][p]]\n",
    "                                    \n",
    "    return e"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "1487e735",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A function to delete keys of those AMR genes who are not present in atleast 25% of the total genomes ##\n",
    "def delete_keys_lessthan_25percent_instances(dict_element):\n",
    "    total_genomes= len(gbkfiles)\n",
    "    minimum_genomes = int((total_genomes /25)*100)\n",
    "    emptykeyslist=[]\n",
    "    for i,j in dict_element.items():\n",
    "        if len(j)<=minimum_genomes:\n",
    "            emptykeyslist.append(i)\n",
    "    for i in emptykeyslist:\n",
    "        del dict_element[i]\n",
    "    return dict_element"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "0d559418",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A function to compute neighbors of each AMR gene based on contig info ##\n",
    "def find_neighbor(rec,uname,data,number_of_genes,drug,genome,key,instancetype):\n",
    "      \n",
    "      neighbor_genes=[]\n",
    "      rec.reset_index(drop=True, inplace=True)\n",
    "      contig_flag=0\n",
    "\n",
    "      for j in range(len(rec['Start'])):\n",
    "            m=[]\n",
    "            n=[]\n",
    "            upwardgenes=[]\n",
    "            downwardgenes=[]\n",
    "            recarray=[]\n",
    "            i=rec.loc[j].Start\n",
    "            w=rec.loc[j].Stop\n",
    "            k=rec.loc[j].req_cont\n",
    "            g=number_of_genes\n",
    "\n",
    "\n",
    "            if k in uname:\n",
    "                newlist=data[k]\n",
    "                newlist.reset_index(drop=True, inplace=True)\n",
    "                for l in range(len(newlist)):\n",
    "                    if newlist['GeneStart'][l] > i and newlist['GeneEnd'][l] > w:\n",
    "                        downwardgenes.append((newlist['GeneStart'][l],newlist['GeneEnd'][l],newlist['GeneStrand'][l],newlist['Locus_Tag'][l],newlist['ProteinSequence'][l],newlist['GeneName'][l],\"#A9F1EE\",\"Not_Applicable\"))\n",
    "                    else:\n",
    "                        upwardgenes.append((newlist['GeneStart'][l],newlist['GeneEnd'][l],newlist['GeneStrand'][l],newlist['Locus_Tag'][l],newlist['ProteinSequence'][l],newlist['GeneName'][l],\"#A9F1EE\",\"Not_Applicable\"))\n",
    "                    #print(upwardgenes) \n",
    "\n",
    "                newu=pd.DataFrame(upwardgenes,columns=[\"GeneStart\",\"GeneEnd\",\"Strand\",\"Locus_Tag\",\"ProteinSequence\",\"GeneName\",\"Genecolor\",\"Gene_Cut_Off\"])\n",
    "                m=(newu.iloc[(newu['GeneStart']-i).abs().argsort()[:g+1]]).sort_values(by=\"GeneStart\")\n",
    "                #print(m)\n",
    "                newd=pd.DataFrame(downwardgenes,columns=[\"GeneStart\",\"GeneEnd\",\"Strand\",\"Locus_Tag\",\"ProteinSequence\",\"GeneName\",\"Genecolor\",\"Gene_Cut_Off\"])\n",
    "                n=(newd.iloc[(newd['GeneStart']-i).abs().argsort()[:g]]).sort_values(by=\"GeneStart\")\n",
    "                #print(n)\n",
    "\n",
    "\n",
    "                recarray.append((rec['Start'][j],rec['Stop'][j],rec['Orientation'][j],rec['Locus_Tag'][j],rec['Predicted_Protein'][j],rec['Best_Hit_ARO'][j],\"#ccccff\",rec[\"Cut_Off\"][j]))\n",
    "                o=pd.DataFrame(recarray,columns=[\"GeneStart\",\"GeneEnd\",\"Strand\",\"Locus_Tag\",\"ProteinSequence\",\"GeneName\",\"Genecolor\",\"Gene_Cut_Off\"])\n",
    "\n",
    "\n",
    "            else:\n",
    "                print(k,genome)\n",
    "                print(\"contig does not exist----\"+\"drugclass:\"+drug)\n",
    "                print(rec['Best_Hit_ARO'][j])\n",
    "      \n",
    "      \n",
    "      if len(m)!=0 and len(n)==0:\n",
    "          m.reset_index(drop=True, inplace=True)\n",
    "          m=m.drop([len(m)-1])\n",
    "          e = pd.concat([m,o,n], ignore_index=True)\n",
    "          e.reset_index(drop=True, inplace=True)\n",
    "          neighbor_genes.append(e)\n",
    "              \n",
    "      if len(m)==0 and len(n)!=0:\n",
    "\n",
    "          m.reset_index(drop=True, inplace=True)\n",
    "          n=n.drop([len(n)-1])\n",
    "          e = pd.concat([m,o,n], ignore_index=True)\n",
    "          e.reset_index(drop=True, inplace=True)\n",
    "          neighbor_genes.append(e)\n",
    "         \n",
    "\n",
    "      if(len(m)!=0 and len(n)!=0):\n",
    "        m.reset_index(drop=True, inplace=True)\n",
    "        m=m.drop([len(m)-1])\n",
    "        e = pd.concat([m,o,n], ignore_index=True)\n",
    "        e.reset_index(drop=True, inplace=True)\n",
    "        neighbor_genes.append(e)\n",
    "            \n",
    "      return neighbor_genes,contig_flag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "8ccd6862",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A function to get separate dicts of locus tags,protein sequences and gene names inorder to compare and write to a fasta file####\n",
    "def getrequiredgenes(frame,number_of_genes,drug):\n",
    "    temp_locus_array=[]\n",
    "    temp_protein_array=[]\n",
    "    temp_genename=[]\n",
    "    #print(drug)\n",
    "    for i in frame:\n",
    "        temp_locus_array.append(list(i['Locus_Tag']))\n",
    "        temp_protein_array.append(i['ProteinSequence'])\n",
    "        temp_genename.append((i['GeneName']))\n",
    "    \n",
    "    \n",
    "    locus_to_protein_dict={}\n",
    "    for i in frame:\n",
    "        for j in range(len(i)):\n",
    "            \n",
    "            locus_to_protein_dict[i['Locus_Tag'][j].strip()]=i['ProteinSequence'][j]\n",
    "  \n",
    "    a=temp_locus_array\n",
    "    a=list(itertools.chain.from_iterable(a))\n",
    "   \n",
    "    b=temp_protein_array\n",
    "    b=list(itertools.chain.from_iterable(b))\n",
    "   \n",
    "    c=temp_genename\n",
    "    c=list(itertools.chain.from_iterable(c))\n",
    "    \n",
    " \n",
    "    if number_of_genes==10:\n",
    "        if len(frame)>1:\n",
    "            return(a[:10]+a[-10:], b[:10]+b[-10:],c[:10]+c[-10:])## for more than one card genes in a genomes\n",
    "        else:\n",
    "            return a,b,c\n",
    "        #return(a[:5]+a[-5:], b[:5]+b[-5:],c[:5]+c[-5:])## for more than one card genes in a genomes\n",
    "    if number_of_genes==14:\n",
    "        \n",
    "        if len(frame)>1:\n",
    "            print(drug)\n",
    "            return(a[:14]+a[-14:], b[:14]+b[-14:],c[:14]+c[-14:])## for more than one card genes in a genomes\n",
    "        else:\n",
    "            return a,b,c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "86729d13",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A function to manipulate contig to compare whether the RGI and GBK genes belong to same contig#####\n",
    "\n",
    "def locus_generator(frame,genome_name):\n",
    "    tag=[]\n",
    "    RGI_name_array=[]\n",
    "        \n",
    "    frame.reset_index(drop=True, inplace=True)\n",
    "    for index in range(len(frame)):\n",
    "        temp_name=frame[\"Best_Hit_ARO\"][index].split(\" \")\n",
    "        if len(temp_name)>1:\n",
    "            name=(temp_name[0][0]+temp_name[1][0]+\"_\"+temp_name[2])\n",
    "            RGI_name_array.append(name)\n",
    "        else:\n",
    "            name=(temp_name[0])\n",
    "            RGI_name_array.append(name)\n",
    "            \n",
    "        tag.append(genome_name+\"(\"+name+\")\"+frame[\"Cut_Off\"][index][0]+\"_\"+str(frame[\"Best_Identities\"][index]))\n",
    "        \n",
    "    return tag,RGI_name_array"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "7f0f8ddf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read each rgi file(.txt) using pandas and store them in a dataframe#\n",
    "dataframelist=[]\n",
    "filenames=[]\n",
    "datadict={}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "51510b6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in sorted(readfiles): \n",
    "    dfak=pd.read_csv(i,sep=\"\\t\") # added by ak to replace '/' by '-'\n",
    "    dfak[\"Best_Hit_ARO\"] = dfak[\"Best_Hit_ARO\"].apply(lambda x: x.replace(\"/\", \"-\"))\n",
    "    filenames.append(os.path.basename(i).split(\".\")[0])\n",
    "    dataframelist.append(pd.DataFrame(dfak))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "52b0757d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dictionary of dataframes with keys as filenames ###\n",
    "for i in range(len(filenames)):\n",
    "    datadict[filenames[i]]=pd.DataFrame(dataframelist[i])  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "9bd593d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replacing the orientation of rgi dataframe to match the data in .gbk format to help with comparison #\n",
    "for i,j in datadict.items():\n",
    "    j[\"Orientation\"]=j[\"Orientation\"].replace(\"-\",\"-1\")\n",
    "    j[\"Orientation\"]=j[\"Orientation\"].replace(\"+\",\"+1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "41f52b6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Adding two new columns 1> Modified Locus tag and 2> contig to extract the right neighbors #\n",
    "for k,v in datadict.items():\n",
    "    newcon=[]\n",
    "    for i in v[\"Contig\"]:\n",
    "        head,sep,tail=i.partition(\"_\")\n",
    "        newcon.append(head)\n",
    "    v['req_cont']=newcon    #adding a new column \"req_cont\" into dataframe\n",
    "    v['Locus_Tag'],v[\"Best_Hit_ARO\"]=locus_generator(v,k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "24dd8d6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read the .gnk file data, store the info in the sorted order in dictionaries #\n",
    "gbk_names=[]\n",
    "uniquenames=[]\n",
    "datasetslist=[]\n",
    "gbkdict={}\n",
    "uniquedict={}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "d1322243",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in sorted(gbkfiles):    \n",
    "    gbk_names.append(os.path.basename(i).split(\".\")[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "7133f5be",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in gbk_names:\n",
    "    uniquenames.append(str(i))\n",
    "    datasetslist.append(str(i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "b98f5e6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "gbkfiles=sorted(gbkfiles) #read the files in the sorted order\n",
    "\n",
    "for i in range(len(gbk_names)):\n",
    "    a=gbk_names[i]\n",
    "    b=uniquenames[i]\n",
    "    gbkdict[a],uniquedict[b]=extract(gbkfiles[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "d9de1b13",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert the Locus tag column from  ['SA200015_0001'] --> SA200015_001 to help while writing to fasta file required to BLAST #\n",
    "for j,i in gbkdict.items():\n",
    "    i['Locus_Tag']=i['Locus_Tag'].apply(lambda i:str(i).replace(\"[\",\"\").replace(\"]\",\"\").replace(\"'\",\"\"))\n",
    "    i['ProteinSequence']=i['ProteinSequence'].str.strip('[]')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "7917b066",
   "metadata": {},
   "outputs": [],
   "source": [
    "# To supress the warning that arise when a single value of dataframe column is manipulated #\n",
    "def fxn():\n",
    "    warnings.warn(\"deprecated\", DeprecationWarning)\n",
    "\n",
    "with warnings.catch_warnings():\n",
    "    warnings.simplefilter(\"ignore\")\n",
    "    fxn()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "a860c3d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('mode.chained_assignment', None)\n",
    "# The block that causes the chained assignment error #\n",
    "\n",
    "for i,j in gbkdict.items():\n",
    "    for value in range(len(j[\"GeneName\"])):\n",
    "        if isinstance(j[\"GeneName\"].iloc[value], list):\n",
    "            j[\"GeneName\"].iloc[value] = j[\"GeneName\"].iloc[value][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "263d8959",
   "metadata": {},
   "outputs": [],
   "source": [
    "# For each .gbk file, divides the data based on available contigs using groupby and dictionary, used to find the neighbors of same contig# \n",
    "datasetdict={}\n",
    "for i in range(len(gbk_names)):\n",
    "    for j,k in gbkdict.items():\n",
    "        datasetdict[j]=make_groups(k,\"contig_name\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "c0e9ff0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Finds the union of all the AMR genes present in more than 25 percent of the total genomes #\n",
    "uniquedrugdict={}\n",
    "unionofdrugclasess=[]\n",
    "\n",
    "for j,k in datadict.items():\n",
    "    uniquedrugclasses=[]\n",
    "    for l in range(len(k)):\n",
    "        if k['Best_Hit_ARO'][l]  not in uniquedrugclasses:\n",
    "                uniquedrugclasses.append(k[\"Best_Hit_ARO\"][l])\n",
    "    uniquedrugdict[j]=uniquedrugclasses\n",
    "        \n",
    "for i,j in uniquedrugdict.items():\n",
    "    for item in j:\n",
    "        if item not in unionofdrugclasess:\n",
    "            unionofdrugclasess.append(item)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "0b52c199",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Converts the longer AMR gene names into shorter version to make them as dictionary keys and to make it easier to visualize in gene order image#\n",
    "# Haemophilis influenzea PBP3 --> Hi_PBP3#\n",
    "\n",
    "listofdrugnames_modified=[]\n",
    "for k in unionofdrugclasess:\n",
    "    if len(k.split(\" \"))>1:\n",
    "        temp=k.split(\" \")\n",
    "        listofdrugnames_modified.append(temp[0][0]+temp[1][0]+\"_\"+temp[2])\n",
    "    else:\n",
    "\n",
    "        listofdrugnames_modified.append(k.split(\"; \")[0].split(\" \")[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "9110aac7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A main dictionary of all the AMR models with all the genomes \n",
    "main_dictionary={}\n",
    "for i in range(len(listofdrugnames_modified)):\n",
    "    main_dictionary[listofdrugnames_modified[i]]=createeachdict_drug(unionofdrugclasess[i])\n",
    "#main_dictionary=delete_keys_lessthan_25percent_instances(main_dictionary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "7725f10b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Divide the AMR genes into single instance and multiple instance #            \n",
    "Dict_multigene_instances={}\n",
    "Dict_singlegene_instances={}\n",
    "for i,j in main_dictionary.items():\n",
    "    temp={}\n",
    "    flag=0\n",
    "    for a,b in j.items():\n",
    "        \n",
    "        if len(b)>1:\n",
    "            flag=1\n",
    "    if flag==1:\n",
    "        Dict_multigene_instances[i]=j\n",
    "    else:\n",
    "        Dict_singlegene_instances[i]=j"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "db079ec9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Code to get one to one genome comparisons when there are multiple instances and divide the frames based on locus tags  #\n",
    "dict_rgi_multiple_occurance={}\n",
    "g={}\n",
    "for i,j in Dict_multigene_instances.items():\n",
    "    temp={}\n",
    "    l=[]\n",
    "    deletekeylist=[]\n",
    "    for a,b in j.items():\n",
    "        if len(b)>1 or len(b)==1:\n",
    "            l.append(a)\n",
    "            b.reset_index(drop=True, inplace=True)            \n",
    "            x=make_groups(b,\"Locus_Tag\")\n",
    "            for e,f in x.items():\n",
    "                temp[e]=f\n",
    "\n",
    "    dict_rgi_multiple_occurance[i]=temp\n",
    "    g[i]=l\n",
    " \n",
    "for i,j in Dict_multigene_instances.items():\n",
    "    for t in g[i]:\n",
    "        del j[t]\n",
    "        \n",
    "for i,j in dict_rgi_multiple_occurance.items():    \n",
    "     j.update(Dict_multigene_instances[i])\n",
    "\n",
    "\n",
    "dict_rgi_single_occurance={}\n",
    "g={}\n",
    "for i,j in Dict_singlegene_instances.items():\n",
    "    temp={}\n",
    "    l=[]\n",
    "    deletekeylist=[]\n",
    "    for a,b in j.items():\n",
    "        if len(b)==1:\n",
    "            l.append(a)\n",
    "            b.reset_index(drop=True, inplace=True)            \n",
    "            x=make_groups(b,\"Locus_Tag\")\n",
    "            for e,f in x.items():\n",
    "                temp[e]=f\n",
    "\n",
    "    dict_rgi_single_occurance[i]=temp\n",
    "    g[i]=l\n",
    " \n",
    "for i,j in Dict_singlegene_instances.items():\n",
    "    for t in g[i]:\n",
    "        del j[t]\n",
    "        \n",
    "for i,j in dict_rgi_single_occurance.items():    \n",
    "     j.update(Dict_singlegene_instances[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "6ef5bc7b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "133 ERR375423(vanRA)S_100.0\n",
      "contig does not exist----drugclass:vanRA\n",
      "vanRA\n",
      "169 SRR14010935(vanRA)S_100.0\n",
      "contig does not exist----drugclass:vanRA\n",
      "vanRA\n",
      "177 SRR14011025(vanRA)S_100.0\n",
      "contig does not exist----drugclass:vanRA\n",
      "vanRA\n",
      "159 ERR369976(dfrF)S_100.0\n",
      "contig does not exist----drugclass:dfrF\n",
      "dfrF\n",
      "146 ERR374958(dfrF)S_100.0\n",
      "contig does not exist----drugclass:dfrF\n",
      "dfrF\n",
      "150 ERR375118(dfrF)S_100.0\n",
      "contig does not exist----drugclass:dfrF\n",
      "dfrF\n",
      "151 ERR375134(dfrF)S_100.0\n",
      "contig does not exist----drugclass:dfrF\n",
      "dfrF\n",
      "169 ERR375445(dfrF)S_100.0\n",
      "contig does not exist----drugclass:dfrF\n",
      "dfrF\n",
      "215 ERR375471(dfrF)S_100.0\n",
      "contig does not exist----drugclass:dfrF\n",
      "dfrF\n",
      "67 ERR1100603(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "38 ERR1156191(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "68 ERR1156217(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "67 ERR1156236(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "79 ERR987638(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "66 SRR14024972(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "65 SRR14024978(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "58 SRR14024983(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "97 SRR14024984(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "71 SRR14024990(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "117 SRR14025000(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "52 SRR14026459(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "46 SRR14026470(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "59 SRR14026532(poxtA)S_100.0\n",
      "contig does not exist----drugclass:poxtA\n",
      "poxtA\n",
      "210 ERR1100577(MexB)S_100.0\n",
      "contig does not exist----drugclass:MexB\n",
      "MexB\n",
      "226 ERR1100631(MexB)S_100.0\n",
      "contig does not exist----drugclass:MexB\n",
      "MexB\n",
      "391 ERR374849(AAC(6')-Iak)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Iak\n",
      "AAC(6')-Iak\n",
      "275 ERR375553(AAC(6')-Iak)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Iak\n",
      "AAC(6')-Iak\n",
      "309 ERR984997(AAC(6')-Iak)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Iak\n",
      "AAC(6')-Iak\n",
      "247 ERR987597(AAC(6')-Iak)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Iak\n",
      "AAC(6')-Iak\n",
      "391 ERR375154(LpeA)S_100.0\n",
      "contig does not exist----drugclass:LpeA\n",
      "LpeA\n",
      "385 ERR375164(LpeA)S_100.0\n",
      "contig does not exist----drugclass:LpeA\n",
      "LpeA\n",
      "427 ERR375156(NDM-10)S_100.0\n",
      "contig does not exist----drugclass:NDM-10\n",
      "NDM-10\n",
      "326 ERR375427(OXA-835)S_100.0\n",
      "contig does not exist----drugclass:OXA-835\n",
      "OXA-835\n",
      "303 ERR375476(AAC(6')-Il)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Il\n",
      "AAC(6')-Il\n",
      "314 ERR375585(EreA2)S_100.0\n",
      "contig does not exist----drugclass:EreA2\n",
      "EreA2\n"
     ]
    }
   ],
   "source": [
    "# Create separate single instance multiple instance folders\n",
    "# Find the neighbors and store them in separate dictionaries\n",
    "\n",
    "singleinstance_neighboringdict_combined_range10={}\n",
    "Single_contig_end_flag_dict={}\n",
    "for i,j in dict_rgi_single_occurance.items():\n",
    "    Dict_neighboring_genes_range10={}\n",
    "    temp_contig_flag={}\n",
    "    for k,l in j.items():\n",
    "        key=k.split(\"(\")[0]\n",
    "        temp,contigflag=find_neighbor(j[k],uniquedict[key],datasetdict[key],10,i,k,key,\"Single_Instance_Neighborhood\")\n",
    "        if len(temp)>0:\n",
    "            Dict_neighboring_genes_range10[k]=temp\n",
    "            temp_contig_flag[k]=contigflag\n",
    "    singleinstance_neighboringdict_combined_range10[i]=Dict_neighboring_genes_range10\n",
    "    Single_contig_end_flag_dict[i]=temp_contig_flag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "8163dd02",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "164 ERR985045(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "139 ERR985080(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "142 SRR14010935(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "175 SRR14010938(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "180 SRR14010943(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "166 SRR14010945(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "160 SRR14010946(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "141 SRR14010960(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "148 SRR14010969(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "153 SRR14010970(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "166 SRR14010985(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "153 SRR14010993(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "152 SRR14011002(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "156 SRR14011006(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "150 SRR14011025(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "142 SRR14011032(Td_reductase)S_41.03\n",
      "contig does not exist----drugclass:Td_reductase\n",
      "Td_reductase\n",
      "187 ERR1036028(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "191 ERR1036028(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "179 ERR1036028(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "103 ERR1036044(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "127 ERR1036044(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "112 ERR1036044(tetM)S_96.94\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "111 ERR1036057(tetM)S_96.94\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "185 ERR1036059(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "174 ERR1036060(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "227 ERR1036070(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "224 ERR1069046(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR1069046(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "102 ERR1100588(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "120 ERR1100588(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "137 ERR1100588(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "88 ERR1100588(tetM)S_97.03\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "90 ERR1100594(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "102 ERR1100594(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "128 ERR1100594(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "80 ERR1100594(tetM)S_97.03\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "76 ERR1100596(tetM)S_96.94\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "131 ERR1100598(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "114 ERR1100599(tetM)S_98.52\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "158 ERR1100625(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "185 ERR1100625(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "160 ERR1100632(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "161 ERR1100632(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "174 ERR1100633(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "207 ERR1100633(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "157 ERR1100633(tetM)S_98.28\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "73 ERR1156197(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "86 ERR1156197(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "76 ERR1156197(tetM)S_96.94\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "154 ERR1156206(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "108 ERR1156207(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "109 ERR1156222(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "119 ERR1156223(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "88 ERR1156231(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "107 ERR1156231(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "99 ERR1156231(tetM)S_96.94\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "131 ERR1156254(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "154 ERR1156254(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "157 ERR1156254(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "97 ERR1156258(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "92 ERR1156264(tetM)S_96.51\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "104 ERR1156268(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "96 ERR1156268(tetM)S_96.51\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "93 ERR1156320(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "86 ERR1156320(tetM)S_96.51\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "150 ERR1216304(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "154 ERR1216304(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "144 ERR1216304(tetM)S_97.82\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "201 ERR369941(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "183 ERR369941(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "241 ERR369951(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "207 ERR369951(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "38 ERR369955(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "46 ERR369955(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "39 ERR369955(tetM)S_96.94\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "230 ERR369958(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "310 ERR369958(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "200 ERR369958(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "175 ERR369964(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "157 ERR369964(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "215 ERR369969(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR369969(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "187 ERR369972(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "196 ERR369972(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "155 ERR369976(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "143 ERR369976(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "186 ERR369978(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "174 ERR369978(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "194 ERR369980(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR369980(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "242 ERR369992(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "207 ERR369992(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "192 ERR369996(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "177 ERR369996(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "215 ERR369997(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "193 ERR369997(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "166 ERR369998(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "152 ERR369998(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "167 ERR370003(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR370003(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "185 ERR370005(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "265 ERR370007(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "379 ERR370007(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "228 ERR370007(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "176 ERR370015(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "164 ERR370015(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "194 ERR370019(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "183 ERR370019(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "237 ERR370021(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "209 ERR370021(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "179 ERR370023(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "187 ERR370023(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "210 ERR370029(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "214 ERR370029(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "188 ERR370029(tetM)S_98.57\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "171 ERR370030(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "161 ERR370030(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "183 ERR374662(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "169 ERR374662(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "261 ERR374674(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "231 ERR374674(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "225 ERR374675(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "191 ERR374675(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "246 ERR374679(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "215 ERR374679(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR374683(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "174 ERR374683(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "233 ERR374686(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "203 ERR374686(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "212 ERR374697(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR374697(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "225 ERR374701(tetM)S_97.6\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "199 ERR374701(tetM)S_98.57\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "262 ERR374707(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "286 ERR374707(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "215 ERR374721(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "193 ERR374721(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR374724(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "158 ERR374724(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "229 ERR374743(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR374743(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "181 ERR374744(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "231 ERR374744(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "183 ERR374746(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "175 ERR374746(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "190 ERR374753(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "171 ERR374753(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "243 ERR374754(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "212 ERR374754(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "203 ERR374769(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR374769(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "225 ERR374804(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "198 ERR374804(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "215 ERR374811(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "193 ERR374811(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "204 ERR374813(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR374813(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "226 ERR374826(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "197 ERR374826(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "181 ERR374829(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "170 ERR374829(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR374843(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "259 ERR374843(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "169 ERR374843(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "247 ERR374848(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "218 ERR374848(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "283 ERR374849(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "221 ERR374849(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "240 ERR374855(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "354 ERR374855(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "204 ERR374855(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "235 ERR374856(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "209 ERR374856(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "244 ERR374864(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "211 ERR374864(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR374871(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "169 ERR374871(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "211 ERR374882(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR374882(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "175 ERR374883(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "157 ERR374883(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "204 ERR374894(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "187 ERR374894(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "196 ERR374907(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR374907(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "167 ERR374912(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "154 ERR374912(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "217 ERR374913(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "305 ERR374913(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR374913(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "220 ERR374914(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "199 ERR374914(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "197 ERR374919(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "175 ERR374919(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR374924(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "228 ERR374924(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "165 ERR374924(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR374929(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "158 ERR374929(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "243 ERR374930(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "210 ERR374930(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "218 ERR374938(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "194 ERR374938(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "235 ERR374939(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "210 ERR374939(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "206 ERR374944(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR374944(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "263 ERR374949(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "219 ERR374949(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "213 ERR374965(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "186 ERR374965(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "182 ERR374971(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "171 ERR374971(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "174 ERR374992(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "162 ERR374992(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "267 ERR374993(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "226 ERR374993(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "228 ERR374998(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "306 ERR374998(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "209 ERR375010(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR375010(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "204 ERR375013(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "227 ERR375013(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "205 ERR375015(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "257 ERR375015(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "185 ERR375015(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "267 ERR375018(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "228 ERR375018(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "246 ERR375019(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "354 ERR375019(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "209 ERR375019(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "226 ERR375022(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "205 ERR375022(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "251 ERR375023(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "218 ERR375023(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "287 ERR375027(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "297 ERR375027(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "218 ERR375027(tetM)S_98.57\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "196 ERR375028(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "183 ERR375028(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "181 ERR375030(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "221 ERR375030(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "164 ERR375030(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "191 ERR375031(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "171 ERR375031(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "239 ERR375035(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "204 ERR375035(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "245 ERR375057(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "204 ERR375057(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "200 ERR375062(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "179 ERR375062(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "262 ERR375071(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "221 ERR375071(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "227 ERR375075(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "313 ERR375075(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "197 ERR375075(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "212 ERR375079(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "191 ERR375079(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "221 ERR375083(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR375083(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "175 ERR375091(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "161 ERR375091(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "186 ERR375094(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "236 ERR375094(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "161 ERR375094(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "248 ERR375105(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "216 ERR375105(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "251 ERR375109(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "274 ERR375109(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "182 ERR375114(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "168 ERR375114(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "177 ERR375130(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "160 ERR375130(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "194 ERR375135(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "176 ERR375135(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "260 ERR375138(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "222 ERR375138(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "206 ERR375143(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR375143(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "235 ERR375153(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "337 ERR375153(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "205 ERR375153(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR375157(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "221 ERR375157(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "238 ERR375161(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "208 ERR375161(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "203 ERR375174(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "181 ERR375174(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "194 ERR375188(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "182 ERR375188(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "193 ERR375195(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "176 ERR375195(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "219 ERR375226(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "192 ERR375226(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "247 ERR375246(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "219 ERR375246(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "245 ERR375248(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "210 ERR375248(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "205 ERR375257(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "186 ERR375257(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "263 ERR375311(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "239 ERR375311(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "215 ERR375313(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "193 ERR375313(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "220 ERR375318(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR375318(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "234 ERR375330(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "202 ERR375330(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "250 ERR375333(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "216 ERR375333(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "321 ERR375334(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "210 ERR375334(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "249 ERR375334(tetM)S_99.19\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "176 ERR375335(tetM)S_98.42\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR375335(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "173 ERR375347(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "223 ERR375360(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "232 ERR375360(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "190 ERR375360(tetM)S_98.57\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "222 ERR375365(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "198 ERR375365(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "231 ERR375382(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "281 ERR375382(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "208 ERR375382(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "179 ERR375409(tetM)S_98.42\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "193 ERR375409(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "270 ERR375427(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "279 ERR375427(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "233 ERR375427(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "234 ERR375430(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "205 ERR375430(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "230 ERR375434(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "205 ERR375434(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "234 ERR375436(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "201 ERR375436(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "165 ERR375445(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "154 ERR375445(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "211 ERR375452(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "187 ERR375452(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "207 ERR375471(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "179 ERR375471(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "188 ERR375480(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "174 ERR375480(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "191 ERR375481(tetM)S_98.42\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "205 ERR375481(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "244 ERR375483(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "213 ERR375483(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "258 ERR375494(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "230 ERR375494(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "244 ERR375496(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "208 ERR375496(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR375497(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "167 ERR375497(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "256 ERR375498(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "229 ERR375498(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "253 ERR375506(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "222 ERR375506(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "232 ERR375512(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "202 ERR375512(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "229 ERR375529(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "202 ERR375529(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "224 ERR375544(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "233 ERR375544(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "194 ERR375544(tetM)S_98.57\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "223 ERR375553(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR375553(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "181 ERR375556(tetM)S_98.42\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR375556(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "229 ERR377425(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "205 ERR377425(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "186 ERR377430(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "177 ERR377430(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "177 ERR377437(tetM)S_98.42\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "186 ERR377437(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "230 ERR377438(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "222 ERR377444(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "192 ERR377444(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "191 ERR377450(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "185 ERR377458(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR377458(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "223 ERR377465(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "229 ERR377465(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR377465(tetM)S_98.57\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "176 ERR377469(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "181 ERR377469(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "164 ERR377469(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "211 ERR377474(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "214 ERR377474(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "186 ERR377474(tetM)S_98.57\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "214 ERR377483(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR377483(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "179 ERR377484(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR377485(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "161 ERR377485(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR377501(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "248 ERR388689(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "215 ERR388689(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "201 ERR388691(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "259 ERR388691(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "245 ERR388692(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "216 ERR388692(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "206 ERR388693(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "182 ERR388693(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "237 ERR388701(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "250 ERR388701(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "212 ERR388701(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "252 ERR388715(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "226 ERR388715(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "261 ERR388717(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "269 ERR388717(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "212 ERR388717(tetM)S_98.57\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "204 ERR388718(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "185 ERR388718(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "198 ERR388725(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "185 ERR388725(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "138 ERR984977(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "175 ERR984984(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "163 ERR984984(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "200 ERR984985(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "151 ERR984986(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "183 ERR984987(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR984990(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "173 ERR984990(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "221 ERR984997(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "132 ERR985015(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "170 ERR985022(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "159 ERR985023(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "193 ERR985034(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "249 ERR985039(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "221 ERR985043(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "221 ERR985045(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "208 ERR985047(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR985064(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "187 ERR985068(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "162 ERR985069(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "177 ERR985069(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "190 ERR985069(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "173 ERR985069(tetM)S_96.03\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "191 ERR985074(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "180 ERR985074(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "181 ERR985080(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "161 ERR985082(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "168 ERR985082(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "177 ERR985082(tetM)S_95.54\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "231 ERR985083(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "161 ERR985085(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "182 ERR985089(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "194 ERR985089(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "209 ERR985102(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "188 ERR985113(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "200 ERR985117(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "177 ERR985120(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "218 ERR985123(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "188 ERR985128(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "216 ERR985130(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "198 ERR985133(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "187 ERR985133(tetM)S_97.38\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "198 ERR987508(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "203 ERR987511(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR987514(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "196 ERR987515(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "184 ERR987516(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "188 ERR987516(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR987516(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "189 ERR987519(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR987526(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "195 ERR987527(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "196 ERR987540(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "181 ERR987543(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "176 ERR987547(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "187 ERR987551(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "118 ERR987562(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "198 ERR987585(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR987590(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "176 ERR987590(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "190 ERR987590(tetM)S_95.54\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "216 ERR987592(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "185 ERR987593(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "187 ERR987593(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "171 ERR987593(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "199 ERR987597(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "146 ERR987601(tetM)S_98.46\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "172 ERR987602(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "211 ERR987605(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "207 ERR987615(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "200 ERR987617(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "194 ERR987626(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "198 ERR987639(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "158 ERR987651(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "206 SRR14010935(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "208 SRR14010942(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "204 SRR14011001(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "206 SRR14011025(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "48 SRR14024961(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "60 SRR14024961(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "52 SRR14024961(tetM)S_96.94\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "88 SRR14024972(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "83 SRR14024972(tetM)S_96.51\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "81 SRR14024978(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "77 SRR14024978(tetM)S_96.51\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "52 SRR14024980(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "53 SRR14024980(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "167 SRR14026516(tetM)S_98.57\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "107 SRR14026542(tetM)S_100.0\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "156 SRR14026551(tetM)S_98.47\n",
      "contig does not exist----drugclass:tetM\n",
      "tetM\n",
      "133 ERR375423(vanSA)S_100.0\n",
      "contig does not exist----drugclass:vanSA\n",
      "vanSA\n",
      "157 ERR987590(vanSA)S_100.0\n",
      "contig does not exist----drugclass:vanSA\n",
      "vanSA\n",
      "221 ERR369954(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "206 ERR369969(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "206 ERR369982(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "175 ERR370021(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "185 ERR374701(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "196 ERR374893(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "182 ERR375019(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "164 ERR375313(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "147 ERR375413(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "165 ERR375423(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "172 ERR375430(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "160 ERR375452(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "155 ERR377473(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "157 ERR388718(vanYA)S_100.0\n",
      "contig does not exist----drugclass:vanYA\n",
      "vanYA\n",
      "193 ERR1100565(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "176 ERR1100565(ANT(6)-Ia)S_99.23\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "179 ERR1100577(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "118 ERR1100601(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "120 ERR1100601(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "114 ERR1100602(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "202 ERR1100630(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "192 ERR1100631(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "50 ERR1156186(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "98 ERR1156230(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "106 ERR1156252(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "91 ERR1156267(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "86 ERR1156269(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "89 ERR1156269(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "175 ERR1156290(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "183 ERR1156290(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "169 ERR1156293(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "176 ERR1156293(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "196 ERR369998(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "294 ERR374693(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "273 ERR374957(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "203 ERR388702(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "173 ERR985127(ANT(6)-Ia)S_100.0\n",
      "contig does not exist----drugclass:ANT(6)-Ia\n",
      "ANT(6)-Ia\n",
      "133 ERR1036065(ErmB)S_98.29\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "157 ERR1100572(ErmB)S_98.67\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "65 ERR1156203(ErmB)S_100.0\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "56 ERR1156203(ErmB)S_98.68\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "138 ERR1156206(ErmB)S_99.11\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "104 ERR1156233(ErmB)S_98.67\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "104 ERR1156248(ErmB)S_98.67\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "72 ERR1156253(ErmB)S_100.0\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "192 ERR375281(ErmB)S_97.39\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "179 ERR375406(ErmB)S_98.25\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "182 ERR375476(ErmB)S_98.25\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "167 ERR388702(ErmB)S_98.25\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "172 ERR985073(ErmB)S_98.25\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "170 ERR985075(ErmB)S_98.25\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "181 ERR985076(ErmB)S_98.25\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "132 ERR985127(ErmB)S_98.67\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "154 ERR987590(ErmB)S_98.25\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "97 SRR14022735(ErmB)S_98.46\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "108 SRR14022735(ErmB)S_99.13\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "71 SRR14022762(ErmB)S_98.46\n",
      "contig does not exist----drugclass:ErmB\n",
      "ErmB\n",
      "154 ERR1036028(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "203 ERR1036028(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "147 ERR1036058(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "168 ERR1036058(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "186 ERR1036059(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "198 ERR1036059(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "215 ERR1036059(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "229 ERR1036059(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "133 ERR1036061(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "157 ERR1036061(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "91 ERR1100601(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "144 ERR369966(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "233 ERR369966(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "150 ERR369981(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "243 ERR369981(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "178 ERR369993(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "227 ERR369993(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "263 ERR370000(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "179 ERR370011(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "236 ERR370011(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "145 ERR370018(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "122 ERR370033(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "151 ERR374663(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "234 ERR374663(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "119 ERR374685(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "192 ERR374693(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "270 ERR374693(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "119 ERR374723(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "159 ERR374727(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "161 ERR374735(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "211 ERR374735(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "123 ERR374752(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "159 ERR374760(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "255 ERR374760(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "190 ERR374770(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "243 ERR374770(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "249 ERR374770(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "131 ERR374774(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "174 ERR374774(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "168 ERR374793(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "295 ERR374793(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "170 ERR374806(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "209 ERR374806(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "116 ERR374824(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "157 ERR374824(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "139 ERR374828(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "240 ERR374828(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "134 ERR374835(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "165 ERR374899(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "245 ERR374899(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "150 ERR374909(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "237 ERR374909(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "141 ERR374916(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "149 ERR374921(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "197 ERR374922(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "286 ERR374922(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "165 ERR374925(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "216 ERR374925(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "138 ERR374933(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "179 ERR374933(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "125 ERR374952(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "150 ERR374959(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "199 ERR374959(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "142 ERR374963(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "214 ERR374963(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "191 ERR374968(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "123 ERR374982(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "250 ERR374986(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "136 ERR374990(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "129 ERR374991(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "176 ERR374991(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "117 ERR374995(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "170 ERR375005(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "129 ERR375014(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "177 ERR375014(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "147 ERR375035(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "253 ERR375035(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "127 ERR375046(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "168 ERR375046(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "145 ERR375051(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "136 ERR375060(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "188 ERR375060(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "134 ERR375066(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "197 ERR375066(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "136 ERR375070(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "178 ERR375070(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "126 ERR375077(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "165 ERR375079(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "244 ERR375079(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "154 ERR375090(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "317 ERR375090(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "140 ERR375092(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "261 ERR375092(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "174 ERR375093(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "198 ERR375093(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "211 ERR375093(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "121 ERR375097(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "126 ERR375129(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "128 ERR375133(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "220 ERR375136(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "260 ERR375136(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "167 ERR375143(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "253 ERR375143(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "209 ERR375145(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "344 ERR375145(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "195 ERR375154(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "243 ERR375154(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "120 ERR375160(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "195 ERR375164(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "367 ERR375164(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "155 ERR375287(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "188 ERR375287(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "168 ERR375353(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "297 ERR375353(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "146 ERR375361(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "157 ERR375361(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "135 ERR375392(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "172 ERR375446(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "304 ERR375446(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "159 ERR375448(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "182 ERR375448(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "139 ERR375450(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "172 ERR375450(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "144 ERR375466(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "221 ERR375466(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "170 ERR375522(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "220 ERR375522(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "168 ERR375538(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "205 ERR375538(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "180 ERR375544(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "246 ERR375544(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "267 ERR375544(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "270 ERR375544(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "170 ERR377433(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "229 ERR377433(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "126 ERR377446(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "128 ERR377481(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "162 ERR377497(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "141 ERR388688(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "183 ERR388688(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "160 ERR388701(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "219 ERR388701(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "145 ERR388724(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "162 ERR984978(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "225 ERR984978(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "136 ERR985001(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "163 ERR985001(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "142 ERR985015(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "148 ERR985015(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "179 ERR985015(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "182 ERR985022(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "191 ERR985022(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "245 ERR985022(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "168 ERR985023(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "179 ERR985023(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "222 ERR985023(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "167 ERR985026(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "190 ERR985026(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "166 ERR985035(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "189 ERR985035(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "179 ERR985039(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "234 ERR985039(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "210 ERR985047(AAC(6')-Ie-APH(2'')-Ia)S_99.48\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "180 ERR985081(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "184 ERR985081(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "218 ERR985081(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "172 ERR985085(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "184 ERR985085(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "226 ERR985085(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "196 ERR985094(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "205 ERR985094(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "263 ERR985094(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "143 ERR985109(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "174 ERR985109(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "146 ERR985111(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "174 ERR985111(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "154 ERR985120(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "234 ERR985120(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "146 ERR985121(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "177 ERR985121(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "305 ERR985130(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "140 ERR987509(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "171 ERR987509(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "184 ERR987514(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "195 ERR987514(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "231 ERR987514(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "152 ERR987516(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "175 ERR987516(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "135 ERR987541(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "156 ERR987541(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "136 ERR987544(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "161 ERR987544(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "135 ERR987553(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "195 ERR987553(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "167 ERR987617(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "253 ERR987617(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "136 ERR987627(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "140 ERR987629(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "211 ERR987640(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "217 ERR987640(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "168 ERR987651(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "178 ERR987651(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "219 ERR987651(AAC(6')-Ie-APH(2'')-Ia)S_100.0\n",
      "contig does not exist----drugclass:AAC(6')-Ie-APH(2'')-Ia\n",
      "AAC(6')-Ie-APH(2'')-Ia\n",
      "231 ERR375023(ErmT)S_98.1\n",
      "contig does not exist----drugclass:ErmT\n",
      "ErmT\n",
      "200 ERR375138(ErmT)S_99.02\n",
      "contig does not exist----drugclass:ErmT\n",
      "ErmT\n",
      "79 ERR1100598(tet(W-N-W))S_61.62\n",
      "contig does not exist----drugclass:tet(W-N-W)\n",
      "tet(W-N-W)\n",
      "121 ERR1100632(tet(W-N-W))S_66.99\n",
      "contig does not exist----drugclass:tet(W-N-W)\n",
      "tet(W-N-W)\n",
      "102 ERR1100637(tet(W-N-W))S_67.68\n",
      "contig does not exist----drugclass:tet(W-N-W)\n",
      "tet(W-N-W)\n",
      "76 SRR14022745(tet(W-N-W))S_66.13\n",
      "contig does not exist----drugclass:tet(W-N-W)\n",
      "tet(W-N-W)\n",
      "77 SRR14022745(tet(W-N-W))S_66.13\n",
      "contig does not exist----drugclass:tet(W-N-W)\n",
      "tet(W-N-W)\n",
      "163 ERR1100565(aad(6))S_100.0\n",
      "contig does not exist----drugclass:aad(6)\n",
      "aad(6)\n",
      "169 ERR1100630(aad(6))S_100.0\n",
      "contig does not exist----drugclass:aad(6)\n",
      "aad(6)\n",
      "183 ERR1100630(aad(6))S_100.0\n",
      "contig does not exist----drugclass:aad(6)\n",
      "aad(6)\n",
      "71 ERR1156237(aad(6))S_100.0\n",
      "contig does not exist----drugclass:aad(6)\n",
      "aad(6)\n",
      "170 ERR1156290(aad(6))S_100.0\n",
      "contig does not exist----drugclass:aad(6)\n",
      "aad(6)\n",
      "164 ERR369998(aad(6))S_100.0\n",
      "contig does not exist----drugclass:aad(6)\n",
      "aad(6)\n",
      "163 ERR1100565(SAT-4)S_98.91\n",
      "contig does not exist----drugclass:SAT-4\n",
      "SAT-4\n",
      "169 ERR1100630(SAT-4)S_98.91\n",
      "contig does not exist----drugclass:SAT-4\n",
      "SAT-4\n",
      "71 ERR1156237(SAT-4)S_98.68\n",
      "contig does not exist----drugclass:SAT-4\n",
      "SAT-4\n",
      "170 ERR1156290(SAT-4)S_98.55\n",
      "contig does not exist----drugclass:SAT-4\n",
      "SAT-4\n",
      "151 ERR374820(SAT-4)S_100.0\n",
      "contig does not exist----drugclass:SAT-4\n",
      "SAT-4\n",
      "148 ERR374832(SAT-4)S_100.0\n",
      "contig does not exist----drugclass:SAT-4\n",
      "SAT-4\n",
      "155 ERR374972(SAT-4)S_100.0\n",
      "contig does not exist----drugclass:SAT-4\n",
      "SAT-4\n",
      "169 ERR375209(SAT-4)S_100.0\n",
      "contig does not exist----drugclass:SAT-4\n",
      "SAT-4\n",
      "283 ERR375427(SAT-4)S_97.92\n",
      "contig does not exist----drugclass:SAT-4\n",
      "SAT-4\n",
      "151 ERR374820(APH(3')-IIIa)S_100.0\n",
      "contig does not exist----drugclass:APH(3')-IIIa\n",
      "APH(3')-IIIa\n",
      "169 ERR374820(APH(3')-IIIa)S_100.0\n",
      "contig does not exist----drugclass:APH(3')-IIIa\n",
      "APH(3')-IIIa\n",
      "148 ERR374832(APH(3')-IIIa)S_100.0\n",
      "contig does not exist----drugclass:APH(3')-IIIa\n",
      "APH(3')-IIIa\n",
      "169 ERR375209(APH(3')-IIIa)S_100.0\n",
      "contig does not exist----drugclass:APH(3')-IIIa\n",
      "APH(3')-IIIa\n",
      "110 SRR14022735(APH(3')-IIIa)S_100.0\n",
      "contig does not exist----drugclass:APH(3')-IIIa\n",
      "APH(3')-IIIa\n",
      "71 SRR14022762(APH(3')-IIIa)S_100.0\n",
      "contig does not exist----drugclass:APH(3')-IIIa\n",
      "APH(3')-IIIa\n",
      "163 ERR1100633(tet(L))S_99.44\n",
      "contig does not exist----drugclass:tet(L)\n",
      "tet(L)\n",
      "174 ERR374959(tet(L))S_100.0\n",
      "contig does not exist----drugclass:tet(L)\n",
      "tet(L)\n",
      "130 ERR374732(tetU)S_95.0\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "128 ERR374937(tetU)S_95.0\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "158 ERR374979(tetU)S_95.45\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "169 ERR375018(tetU)S_100.0\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "170 ERR375071(tetU)S_96.0\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "139 ERR375092(tetU)S_96.0\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "134 ERR375161(tetU)S_96.0\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "142 ERR375406(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "142 ERR375476(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "158 ERR375494(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "162 ERR375498(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "147 ERR375542(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "151 ERR377468(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "145 ERR388702(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "180 ERR388723(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "136 ERR985073(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "135 ERR985075(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "141 ERR985076(tetU)S_95.65\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n",
      "113 ERR987544(tetU)S_98.88\n",
      "contig does not exist----drugclass:tetU\n",
      "tetU\n"
     ]
    }
   ],
   "source": [
    "multipleinstance_neighboringdict_combined_range_10={}\n",
    "multiple_contig_end_flag_dict={}\n",
    "for i,j in dict_rgi_multiple_occurance.items():\n",
    "    Dict_neighboring_genes_range10={}\n",
    "    temp_contig_flag={}\n",
    "    for k,l in j.items():\n",
    "        key=k.split(\"(\")[0]\n",
    "        temp,contigflag=find_neighbor(j[k],uniquedict[key],datasetdict[key],10,i,k,key,\"Multiple_Instance_Neighborhood\")\n",
    "        if len(temp)>0:\n",
    "            Dict_neighboring_genes_range10[key]=temp\n",
    "            temp_contig_flag[key]=contigflag\n",
    "    multipleinstance_neighboringdict_combined_range_10[i]=Dict_neighboring_genes_range10\n",
    "    multiple_contig_end_flag_dict[i]=temp_contig_flag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "079e0935",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Code to extract the locus tags, protein sequences and gene names to use to later for comparison #\n",
    "Drug_singleinstance_range_10_locus_dict10={}\n",
    "Drug_singleinstance_range_10_protein_dict10={}\n",
    "Drug_singleinstance_range_10_genename_dict10={}\n",
    "\n",
    "for i,j in singleinstance_neighboringdict_combined_range10.items():\n",
    "    locustags_dict_10={}\n",
    "    protein_dict_10={}\n",
    "    genename_dict_10={}\n",
    "    for k,l in j.items():\n",
    "        locustags_dict_10[k],protein_dict_10[k],genename_dict_10[k]=getrequiredgenes(j[k],10,i)  \n",
    "    Drug_singleinstance_range_10_locus_dict10[i]=locustags_dict_10\n",
    "    Drug_singleinstance_range_10_protein_dict10[i]=protein_dict_10\n",
    "    Drug_singleinstance_range_10_genename_dict10[i]=genename_dict_10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "13fb533e",
   "metadata": {},
   "outputs": [],
   "source": [
    "Drug_multiple_instance_range_10_locus_dict10={}\n",
    "Drug_multiple_instance_range_10_protein_dict10={}\n",
    "Drug_multiple_instance_range_10_genename_dict10={}\n",
    "\n",
    "for i,j in multipleinstance_neighboringdict_combined_range_10.items():\n",
    "\n",
    "    locustags_dict_10={}\n",
    "    protein_dict_10={}\n",
    "    genename_dict_10={}\n",
    "    for k,l in j.items():\n",
    "\n",
    "        locustags_dict_10[k],protein_dict_10[k],genename_dict_10[k]=getrequiredgenes(j[k],10,i)\n",
    "    #print(len(locustags_dict_10),len(protein_dict_10),len(genename_dict_10))  \n",
    "    Drug_multiple_instance_range_10_locus_dict10[i]=locustags_dict_10\n",
    "    Drug_multiple_instance_range_10_protein_dict10[i]=protein_dict_10\n",
    "    Drug_multiple_instance_range_10_genename_dict10[i]=genename_dict_10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "867a955f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read the .txt files of BLAST results and store in a dictionary ######\n",
    "filepath_single='/home/amjad/NICHEnbhd/Outputs/Single_instance/output_blast_single_instance'\n",
    "Single_instance_blastdataframelist=[]\n",
    "Single_instance_blastfilename=[]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "dcd3f87f",
   "metadata": {},
   "outputs": [],
   "source": [
    "readfiles=glob.glob(os.path.join(filepath_single,\"*.txt\"))\n",
    "for i in readfiles:    \n",
    "    Single_instance_blastfilename.append(os.path.basename(i).split(\".\")[0])\n",
    "    i=pd.read_csv(i,index_col = False,sep=\"\\t\", names=['query_id','sub_id','PI','len','Evalue','bitscore','qseq','sseq'])\n",
    "    Single_instance_blastdataframelist.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "e95d77bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "temp_Single_instance_blastdatadict={}### dictionary of dataframes with keys as filenames ###\n",
    "for i in range(len(Single_instance_blastfilename)):\n",
    "    a=Single_instance_blastdataframelist[i]\n",
    "    b=Single_instance_blastfilename[i]\n",
    "    temp_Single_instance_blastdatadict[b]=pd.DataFrame(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "7be84ac9",
   "metadata": {},
   "outputs": [],
   "source": [
    "filepath_multiple='/home/amjad/NICHEnbhd/Outputs/Multiple_instance/output_blast_multiple_instance'\n",
    "Multiple_instance_blastdataframelist=[]\n",
    "Multiple_instance_blastfilename=[]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4085f45d",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'glob' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_562237/3595403488.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mreadfiles\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mglob\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mglob\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_multiple\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"*.txt\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mreadfiles\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0mMultiple_instance_blastfilename\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbasename\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\".\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mi\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mindex_col\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"\\t\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnames\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'query_id'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'sub_id'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'PI'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'len'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'Evalue'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'bitscore'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'qseq'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'sseq'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0;31m#i=pd.read_csv(i,index_col = False,sep=\"\\t\")\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'glob' is not defined"
     ]
    }
   ],
   "source": [
    "readfiles=glob.glob(os.path.join(filepath_multiple,\"*.txt\"))\n",
    "for i in readfiles:\n",
    "    Multiple_instance_blastfilename.append(os.path.basename(i).split(\".\")[0])\n",
    "    i=pd.read_csv(i,index_col = False,sep=\"\\t\", names=['query_id','sub_id','PI','len','Evalue','bitscore','qseq','sseq'])\n",
    "    #i=pd.read_csv(i,index_col = False,sep=\"\\t\")\n",
    "    Multiple_instance_blastdataframelist.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1790dcc3",
   "metadata": {},
   "outputs": [],
   "source": [
    "temp_Multiple_instance_blastdatadict={}### dictionary of dataframes with keys as filenames ###\n",
    "for i in range(len(Multiple_instance_blastfilename)):\n",
    "    a=Multiple_instance_blastdataframelist[i]\n",
    "    b=Multiple_instance_blastfilename[i]\n",
    "    temp_Multiple_instance_blastdatadict[b]=pd.DataFrame(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc55d31f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# To sort and filter only those BLAST results with more than 70 percent identity \n",
    "\n",
    "Single_instance_blastdatadict={}\n",
    "for i,j in temp_Single_instance_blastdatadict.items():\n",
    "    j=j[j[\"PI\"]>70]\n",
    "    Single_instance_blastdatadict[i]=j"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fbfcccfa",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A function to normalize the bitscore values \n",
    "def normalized_bitscore(z): \n",
    "  manval=[]\n",
    "  a=z[z['query_id']==z['sub_id']]\n",
    "  manval=[]\n",
    "  a=z[z['query_id']==z['sub_id']]\n",
    "  a.reset_index(drop=True, inplace=True)\n",
    "  q={}\n",
    "  for i in range(len(a)):\n",
    "    q[a['query_id'][i]]=a['bitscore'][i]\n",
    "\n",
    "  for i in range(len(z)):\n",
    "      manval.append(round(float(z['bitscore'][i])/q[z['query_id'][i]],3))\n",
    "\n",
    "  z['normalized_bitscore']=manval\n",
    "  return z"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ac3511fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "Single_instance_drugclass_genomedict={}\n",
    "for k,l in Drug_singleinstance_range_10_locus_dict10.items():\n",
    "    u=Single_instance_blastdatadict[k]\n",
    "    temp_dict={}\n",
    "    for a,b in l.items():        \n",
    "        temp=[]\n",
    "        u.reset_index(drop=True, inplace=True)\n",
    "        for index in range (len(u)):\n",
    "            if  u[\"query_id\"][index] in b:\n",
    "                temp.append((u[\"query_id\"][index],u[\"sub_id\"][index],u[\"PI\"][index],u[\"bitscore\"][index]))\n",
    "                    \n",
    "        #frame=pd.DataFrame(temp,columns=[\"query_id\",\"sub_id\",\"PI\",\"bitscore\"])\n",
    "        temp_dict[a]=pd.DataFrame(temp,columns=[\"query_id\",\"sub_id\",\"PI\",\"bitscore\"])\n",
    "      \n",
    "    Single_instance_drugclass_genomedict[k]=temp_dict\n",
    "    \n",
    "\n",
    "for i,j in Single_instance_drugclass_genomedict.items():\n",
    "    for a,b in j.items():\n",
    "        b=normalized_bitscore(b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "865f965e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A function to remove BLAST entry results for same gene comparison by taking the highest bit score \n",
    "def remove_duplicates(temp):\n",
    "    df=pd.DataFrame(temp,columns=[\"query_id\",\"sub_id\",\"bitscore\"])\n",
    "    d = df.sort_values(\"bitscore\", ascending=False)\n",
    "    d = d.drop_duplicates(['query_id'])\n",
    "    d.reset_index(drop=True, inplace=True)\n",
    "    \n",
    "    return d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c9021c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "single_similarity_array_dict={}\n",
    "for k,l in Drug_singleinstance_range_10_locus_dict10.items():\n",
    "    similarity_array=[]\n",
    "    for genome1 in Single_instance_drugclass_genomedict[k].keys():  \n",
    "        u=Single_instance_drugclass_genomedict[k][genome1]\n",
    "        for genome2,b in l.items():\n",
    "            temp=[]\n",
    "            sum1=0\n",
    "            sum2=0\n",
    "            u.reset_index(drop=True, inplace=True)\n",
    "            for index in range (len(u)):\n",
    "                if u[\"sub_id\"][index] in b:\n",
    "                    temp.append((u[\"query_id\"][index],u[\"sub_id\"][index],u[\"normalized_bitscore\"][index]))\n",
    "\n",
    "            tempframe=remove_duplicates(temp)\n",
    "            sum1=round(tempframe[\"bitscore\"].sum(),3)\n",
    "            \n",
    "            if Single_contig_end_flag_dict[k][genome1] ==1:\n",
    "                length_original_neighborhood=len(Drug_singleinstance_range_10_locus_dict10[k][genome1])\n",
    "                difference=length_original_neighborhood-len(tempframe)\n",
    "                sum2= sum1+(21-difference-len(tempframe))\n",
    "                \n",
    "            elif Single_contig_end_flag_dict[k][genome2]==1:\n",
    "                length_original_neighborhood=len(Drug_singleinstance_range_10_locus_dict10[k][genome2])\n",
    "                difference=length_original_neighborhood-len(tempframe)                                 \n",
    "                \n",
    "                sum2= sum1+(21-difference-len(tempframe))\n",
    "            else:\n",
    "                sum2=sum1\n",
    "           \n",
    "            similarity_array.append((genome1,genome2,round(sum2,3)))\n",
    "    single_similarity_array_dict[k]=similarity_array"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1250b57",
   "metadata": {},
   "outputs": [],
   "source": [
    "# A function to convert the similarity matrix to symmteric matric matrix by taking the average of scores\n",
    "def converttosymmetrix(j):\n",
    "    newj=[]\n",
    "    \n",
    "    for index1 in range(len(j)):\n",
    "        \n",
    "        score1=j[index1][2]\n",
    "        for index2 in range(len(j)):\n",
    "            if j[index2][1]==j[index1][0] and j[index2][0]==j[index1][1]:\n",
    "                avg=round((j[index1][2]+j[index2][2])/2,3)\n",
    "        newj.append((j[index1][0],j[index1][1],avg))\n",
    "    return newj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84ef2d14",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use the function to convert to symmetric matrix and store the final matrix in separate dictionaries\n",
    "Single_instance_matrixframe={}\n",
    "for i,j in single_similarity_array_dict.items():\n",
    "    new=converttosymmetrix(j)\n",
    "    finalresult=pd.DataFrame(new,columns=[\"Query_id\",\"Sub_id\",\"normalized_bitscore\"])\n",
    "    sim_mat=pd.crosstab(index=finalresult.iloc[:,0], columns=finalresult.iloc[:,1],values=finalresult.iloc[:,2], aggfunc=lambda x: x,colnames=None)\n",
    "    Y=sim_mat.values\n",
    "    Single_instance_matrixframe[i]=sim_mat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "87d8a48a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generate a distance matrix and generate UPGMA clusters \n",
    "os.mkdir(\"Outputs/UPGMA_single_clusters\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ad4bba2",
   "metadata": {},
   "outputs": [],
   "source": [
    "save_path=\"Outputs/UPGMA_single_clusters\"\n",
    "dataarray=[]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3f28226",
   "metadata": {},
   "outputs": [],
   "source": [
    "hierarchy.set_link_color_palette(['r', 'g', 'y', 'm'])\n",
    "matplotlib.rcParams['lines.linewidth'] = 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aadf8d76",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i,j in Single_instance_matrixframe.items(): \n",
    "    if len(j)>1:\n",
    "\n",
    "        df = pd.DataFrame(j,columns=j.keys())\n",
    "        r  = df.values\n",
    "        df = df.transform(lambda x: 1 - x/r.max() )\n",
    "        df = df.round(decimals=3)\n",
    "        Y = df.values\n",
    "        np.fill_diagonal(Y, 0)\n",
    "        Y = distance.squareform(Y)\n",
    "        fig = plt.figure(figsize=(15,5))\n",
    "        Z = hierarchy.linkage(Y, 'average')\n",
    "        dendrogram(Z, leaf_rotation=90, leaf_font_size=8, labels=j.keys())\n",
    "        savename=os.path.join(save_path,i+\"_.png\")\n",
    "        plt.ylabel(\"Distance between neighborhoods\")\n",
    "        plt.xlabel(\"Neighborhoods with their respective Genome IDs\")\n",
    "        plt.savefig(savename,bbox_inches='tight', dpi=100)\n",
    "        plt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab281270",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Execute if only there are multiple instance AMR gene models #\n",
    "\n",
    "if len(temp_Multiple_instance_blastdatadict.keys())>0:\n",
    "\n",
    "    Multiple_instance_blastdatadict={}\n",
    "    for i,j in temp_Multiple_instance_blastdatadict.items():\n",
    "        j=j[j[\"PI\"]>70]\n",
    "        j = j.reset_index().drop_duplicates(subset=['query_id','sub_id','bitscore'],keep='first').set_index('index')\n",
    "        Multiple_instance_blastdatadict[i]=j\n",
    "\n",
    "\n",
    "    # Divide the mixed BLAST results into their respective AMR gene models and genomes \n",
    "    multiple_instance_drugclass_genomedict={}\n",
    "    for k,l in Drug_multiple_instance_range_10_locus_dict10.items():\n",
    "        u=Multiple_instance_blastdatadict[k]\n",
    "        temp_dict={} \n",
    "        for a,b in l.items():#          \n",
    "            temp=[]\n",
    "            u.reset_index(drop=True, inplace=True)\n",
    "            for index in range (len(u)):\n",
    "                if  u[\"query_id\"][index] in b:\n",
    "                    temp.append((u[\"query_id\"][index],u[\"sub_id\"][index],u[\"PI\"][index],u[\"bitscore\"][index]))\n",
    "                        \n",
    "            #frame=pd.DataFrame(temp,columns=[\"query_id\",\"sub_id\",\"PI\",\"bitscore\"])\n",
    "            temp_dict[a]=pd.DataFrame(temp,columns=[\"query_id\",\"sub_id\",\"PI\",\"bitscore\"])\n",
    "          \n",
    "        multiple_instance_drugclass_genomedict[k]=temp_dict\n",
    "\n",
    "    # using the function and normalizing \n",
    "    for i,j in multiple_instance_drugclass_genomedict.items():\n",
    "        for a,b in j.items():\n",
    "            b=normalized_bitscore(b)\n",
    "\n",
    "\n",
    "    # For each gene model compare the genomes and generate a similarity matrix #\n",
    "    multiple_similarity_array_dict={}\n",
    "    for k,l in Drug_multiple_instance_range_10_locus_dict10.items():\n",
    "        similarity_array=[]\n",
    "        for genome1 in multiple_instance_drugclass_genomedict[k].keys():  \n",
    "            u=multiple_instance_drugclass_genomedict[k][genome1]\n",
    "            for genome2,b in l.items():\n",
    "                temp=[]\n",
    "                sum1=0\n",
    "                sum2=0\n",
    "                u.reset_index(drop=True, inplace=True)\n",
    "                for index in range (len(u)):\n",
    "                    if u[\"sub_id\"][index] in b:\n",
    "                        temp.append((u[\"query_id\"][index],u[\"sub_id\"][index],u[\"normalized_bitscore\"][index]))\n",
    "\n",
    "                tempframe=remove_duplicates(temp)\n",
    "                sum1=round(tempframe[\"bitscore\"].sum(),3)\n",
    "                \n",
    "                if multiple_contig_end_flag_dict[k][genome1] ==1:\n",
    "                    length_original_neighborhood=len(Drug_multiple_instance_range_10_locus_dict10[k][genome1])\n",
    "                    difference=length_original_neighborhood-len(tempframe)\n",
    "                    sum2= sum1+(21-difference-len(tempframe))\n",
    "                    \n",
    "                elif multiple_contig_end_flag_dict[k][genome2]==1:\n",
    "                    length_original_neighborhood=len(Drug_multiple_instance_range_10_locus_dict10[k][genome2])\n",
    "                    difference=length_original_neighborhood-len(tempframe)                                 \n",
    "                    \n",
    "                    sum2= sum1+(21-difference-len(tempframe))\n",
    "                else:\n",
    "                    sum2=sum1\n",
    "               \n",
    "                similarity_array.append((genome1,genome2,round(sum1,3)))\n",
    "        multiple_similarity_array_dict[k]=similarity_array\n",
    "\n",
    "    multiple_instance_matrixframe={}\n",
    "    for i,j in multiple_similarity_array_dict.items():\n",
    "        new=converttosymmetrix(j)\n",
    "        finalresult=pd.DataFrame(new,columns=[\"Query_id\",\"Sub_id\",\"normalized_bitscore\"])\n",
    "        sim_mat=pd.crosstab(index=finalresult.iloc[:,0], columns=finalresult.iloc[:,1],values=finalresult.iloc[:,2], aggfunc=lambda x: x,colnames=None)\n",
    "        Y=sim_mat.values\n",
    "        multiple_instance_matrixframe[i]=sim_mat\n",
    "\n",
    "    os.mkdir(\"Outputs/UPGMA_multiple_clusters\")\n",
    "    save_path=\"Outputs/UPGMA_multiple_clusters\"\n",
    "    dataarray=[]\n",
    "    hierarchy.set_link_color_palette(['r', 'g', 'y', 'm'])\n",
    "    matplotlib.rcParams['lines.linewidth'] = 5\n",
    "    for i,j in multiple_instance_matrixframe.items(): \n",
    "        if len(j)>1:\n",
    "\n",
    "            df = pd.DataFrame(j,columns=j.keys())\n",
    "            r  = df.values\n",
    "            df = df.transform(lambda x: 1 - x/r.max() )\n",
    "            df = df.round(decimals=3)\n",
    "            Y = df.values\n",
    "            np.fill_diagonal(Y, 0)\n",
    "            Y = distance.squareform(Y)\n",
    "            fig = plt.figure(figsize=(20,5))\n",
    "            Z = hierarchy.linkage(Y, 'average')\n",
    "            dendrogram(Z, leaf_rotation=90, leaf_font_size=8, labels=j.keys())\n",
    "            savename=os.path.join(save_path,i+\"_.png\")\n",
    "            plt.ylabel(\"Distance between neighborhoods\")\n",
    "            plt.xlabel(\"Neighborhoods with their respective Genome IDs\")\n",
    "            plt.savefig(savename,bbox_inches='tight', dpi=300)\n",
    "            plt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a094835f",
   "metadata": {},
   "outputs": [],
   "source": [
    "executionTime = (time.time() - startTime)   \n",
    "print('Execution time in seconds: ' + str(executionTime))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
